### PR TITLE
Enable Ant put task to perform filtering.

### PR DESCRIPTION
**Pull request by mscheibler**
I did a small improvement on the Ant Wrapper to enable filtering during a 'put' operation just as ant can do while copying.

### CODE REVIEW

File: src/main/java/com/github/sardine/ant/command/Put.java [hunk 1]
```diff
@@ -39,6 +48,13 @@
 	/** The optional content type of the single source file. */
 	private String fContentType;
 
+	/** Settings for put with filtering. */
+	private FilterSetCollection fFilterSets = new FilterSetCollection();
+	private Vector<FilterChain> fFilterChains = new Vector<FilterChain>();
```
Comment (lookfirst) [inline src/main/java/com/github/sardine/ant/command/Put.java]: I think we stopped using Vector in about 1999. =)

File: src/main/java/com/github/sardine/ant/command/Put.java [hunk 2]
```diff
@@ -39,6 +48,13 @@
...
+	private String fInputEncoding=null;
+	private String fOutputEncoding=null;
+	private boolean fInMemoryFiltering = true;
```
Comment (lookfirst) [inline src/main/java/com/github/sardine/ant/command/Put.java]: why are you prefixing properties with f? ewww...

File: src/main/java/com/github/sardine/ant/command/Put.java [hunk 3]
```diff
@@ -111,7 +127,25 @@ private void checkOrCreateDir(URL baseUrl, File dir, Set<URL> createdDirs) throw
 	 */
 	private void process(File file, URL dest, boolean expectContinue) throws Exception {
 		log("putting " + file + " to " + dest + " with expectContinue=" + expectContinue, Project.MSG_VERBOSE);
-		getSardine().put(dest.toString(), new FileInputStream(file), fContentType, expectContinue);
+		if (fFilterSets.hasFilters() || !fFilterChains.isEmpty()) {
+			Resource filteredResult;
+			File tempFile=null;
+			if (fInMemoryFiltering){
```
Comment (lookfirst) [inline src/main/java/com/github/sardine/ant/command/Put.java]: You should use consistent formatting. `if (...) {` and `} else {`

File: src/main/java/com/github/sardine/ant/command/Put.java (hunk 1 above)
Comment (mscheibler) [inline src/main/java/com/github/sardine/ant/command/Put.java]: Ohh yes, i know!!! :-/ But the Ant ResourceUtils require it.

File: src/main/java/com/github/sardine/ant/command/Put.java (hunk 2 above)
Comment (mscheibler) [inline src/main/java/com/github/sardine/ant/command/Put.java]: A bad habbit due to my companys style guide: 'f' for 'field' has to be before all class properties. If you don't like it i drop it.

File: src/main/java/com/github/sardine/ant/command/Put.java (hunk 3 above)
Comment (mscheibler) [inline src/main/java/com/github/sardine/ant/command/Put.java]: oops, sorry. The auto formatting was turned of

**Comment by mscheibler** [comment]
Stumbled across this old pull request, did a re-base. Maybe it can get integrated now?

**Comment by lookfirst** [comment]
Thanks. =)

**Pull request merged**